# HG changeset patch
# User omajid
# Date 1434565087 14400
#      Wed Jun 17 14:18:07 2015 -0400
# Node ID 44d6e4ff3770e0d272cc5d5e5cf88762e108e2c5
# Parent  d4d7a3107d2039e6077391cf83b70ff7074653b0
8087156, PR2444: SetupNativeCompilation ignores CFLAGS_release for cpp files
Reviewed-by: erikj, ihse

diff -r d4d7a3107d20 -r 44d6e4ff3770 make/common/NativeCompilation.gmk
--- openjdk///make/common/NativeCompilation.gmk	Fri Jun 12 21:25:45 2015 +0100
+++ openjdk///make/common/NativeCompilation.gmk	Wed Jun 17 14:18:07 2015 -0400
@@ -323,6 +323,13 @@
     $1_EXTRA_CXXFLAGS+=$$($1_CXXFLAGS_$(OPENJDK_TARGET_OS)_release)
   endif
 
+  ifeq ($$($1_CXXFLAGS),)
+    $1_CXXFLAGS:=$$($1_CFLAGS)
+  endif
+  ifeq ($$(strip $$($1_EXTRA_CXXFLAGS)),)
+    $1_EXTRA_CXXFLAGS:=$$($1_EXTRA_CFLAGS)
+  endif
+
   ifneq (,$$($1_DEBUG_SYMBOLS))
     ifeq ($(ENABLE_DEBUG_SYMBOLS), true)
       ifdef OPENJDK
@@ -340,13 +347,6 @@
     endif
   endif
 
-  ifeq ($$($1_CXXFLAGS),)
-    $1_CXXFLAGS:=$$($1_CFLAGS)
-  endif
-  ifeq ($$(strip $$($1_EXTRA_CXXFLAGS)),)
-    $1_EXTRA_CXXFLAGS:=$$($1_EXTRA_CFLAGS)
-  endif
-
   ifneq (,$$($1_REORDER))
     $1_EXTRA_CFLAGS += $$(C_FLAG_REORDER)
     $1_EXTRA_CXXFLAGS += $$(CXX_FLAG_REORDER)
